src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -304,15 +304,9 @@ def list_vault(client, resource_group_name=None):
     return list(vault_list)
 
 
-# pylint: disable=inconsistent-return-statements
 def _get_current_user_object_id(graph_client):
-    from azure.cli.command_modules.role import GraphError
-    try:
-        current_user = graph_client.signed_in_user_get()
-        if current_user and current_user.get('id', None):  # pylint:disable=no-member
-            return current_user['id']  # pylint:disable=no-member
```

Comment:
I did leave a comment for this weird behavior of checking the existence of `id` (https://github.com/Azure/azure-cli/pull/22203#discussion_r859747654). 

`graph_client.signed_in_user_get` will ALWAYS return a user object which MUST have `id` property UNLESS `GraphError` is raised.

Takeaways:

1. Pylint error is a sign of abnormality and should NOT be disabled brutally.
2. For anything unusual like `current_user.get('id', None)` or `except CloudError`, comments MUST be left to explain its behavior.

Reply:
We got tripped by some code written almost 6 years ago (cb0faa82bc5db56a15d0c03e4dad5cf040ae9d83)!